src/Microsoft.Unity.Analyzers.Tests/BaseTest.cs

```diff
@@ -70,13 +74,32 @@ public async Task DoNotFailWithInterfaceMembers()
 		await VerifyCSharpDiagnosticAsync(BaseDiagnosticVerifierTest<TAnalyzer>.InterfaceTest);
 	}
 
-	protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
+	protected override TAnalyzer GetCSharpDiagnosticAnalyzer()
 	{
 		return new TAnalyzer();
 	}
 
-	protected override CodeFixProvider GetCSharpCodeFixProvider()
+	protected override TCodeFix GetCSharpCodeFixProvider()
 	{
 		return new TCodeFix();
 	}
+
+	protected bool MethodExists(string typeName, string methodName, string assenblyNameFilter = "")
+	{
+		foreach (var assemblyFile in UnityAssemblies().Where(n => n.Contains(assenblyNameFilter)))
+		{
+			var assembly = Assembly.LoadFile(assemblyFile);
```

Comment:
Doing a full `Assembly.LoadFile` on Unity assembly seems fragile and we're never going to execute code into that. Maybe use a reflection only assembly load context or Cecil?

Reply:
Going to use `AssemblyLoadContext`, `Assembly.ReflectionOnlyLoad` is not a thing anymore in .NET Core.